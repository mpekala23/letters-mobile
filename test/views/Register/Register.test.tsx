import React from 'react';
import { RegisterScreen } from '@views';
import { register } from '@api';
import { render, fireEvent, toJSON, act } from '@testing-library/react-native';
import fetchMock from 'jest-fetch-mock';
import { sleep } from '@utils';

process.env.MOCK_API_IP = 'test_mock_api';

jest.mock('@api', () => ({
  register: jest.fn(),
}));

const setup = (response = {}) => {
  const navigation = { navigate: jest.fn() };
  if (Object.keys(response).length > 0) {
    fetchMock.mockOnce(JSON.stringify(response));
  }
  return render(<RegisterScreen navigation={navigation} />);
};

describe('Register screen', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should have register button be disabled until all fields are valid', async () => {
    jest.useRealTimers();
    const { getByPlaceholderText, getByText } = setup();
    fireEvent.press(getByText('Register'));
    expect(register).toHaveBeenCalledTimes(0);
    fireEvent.changeText(getByPlaceholderText('First Name'), 'Team');
    fireEvent.changeText(getByPlaceholderText('Last Name'), 'Ameelio');
    fireEvent.changeText(
      getByPlaceholderText('Cell Phone Number'),
      '4324324432'
    );
    fireEvent.changeText(getByPlaceholderText('Address Line 1'), 'Somewhere');
    fireEvent.changeText(getByPlaceholderText('Zip Code'), '12345');
    fireEvent.changeText(getByPlaceholderText('City'), 'New Haven');
    fireEvent.changeText(getByPlaceholderText('State'), 'Iowa');
    fireEvent.changeText(
      getByPlaceholderText('E-mail Address'),
      'team@ameelio.org'
    );
    fireEvent.changeText(getByPlaceholderText('Password'), 'ThisGood1');
    fireEvent.changeText(getByPlaceholderText('Confirm Password'), 'ThisGood1');
    fireEvent.changeText(
      getByPlaceholderText('How did you hear about Ameelio?'),
      'Other'
    );
    await act(async () => {
      fireEvent.press(getByText('Register'));
      await sleep(10);
    });
    expect(register).toHaveBeenCalledTimes(1);
  });
});
