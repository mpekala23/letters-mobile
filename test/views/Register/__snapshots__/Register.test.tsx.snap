// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <ScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingBottom": 100,
          "paddingTop": 50,
        }
      }
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 10,
        }
      }
    >
      <View
        accessibilityLabel="Tap to upload profile image"
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Array [
              Object {
                "height": 100,
                "width": 100,
              },
              Object {
                "alignItems": "center",
                "borderColor": "#6E757C",
                "borderRadius": 20,
                "borderWidth": 5,
                "justifyContent": "center",
                "overflow": "hidden",
              },
              Object {},
            ]
          }
          testID="clickable"
        >
          <Text>
            +
          </Text>
        </TouchableOpacity>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontStyle": "italic",
                "fontWeight": "400",
              },
              Object {
                "marginTop": 5,
              },
            ]
          }
        >
          Click to upload profile image.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D0E5FD",
            "borderColor": "#BEDAFC",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 15,
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontWeight": "600",
              },
              Object {
                "color": "#124181",
                "fontSize": 18,
                "fontWeight": "400",
              },
            ]
          }
        >
          We value your privacy. We will only use your address information as the return address for the mailed letter.
        </Text>
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "marginBottom": 10,
          }
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
              },
              Object {
                "color": "#3490dc",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Already have an account?
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.7}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3490dc",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {},
            Object {},
            Object {},
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
              Object {},
            ]
          }
        >
          Dev Skip
        </Text>
      </TouchableOpacity>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="First Name"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Last Name"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 8,
              "marginBottom": 10,
              "overflow": "hidden",
              "width": "100%",
            }
          }
          testID="parent"
        >
          <View
            testID="invalid"
          />
          <View
            testID="unfocused"
          />
          <TextInput
            allowFontScaling={true}
            multiline={false}
            numberOfLines={1}
            placeholder="Cell Phone Number"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderColor": "rgba(0,0,0,0)",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#9A9A9A",
                  "flex": 1,
                  "fontSize": 16,
                  "padding": 10,
                  "textAlignVertical": "top",
                  "width": "100%",
                },
                Object {},
                Object {},
                Object {},
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View />
          <View />
          <View
            style={
              Object {
                "height": 0,
                "width": "100%",
                "zIndex": 5,
              }
            }
          >
            <View />
          </View>
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Address Line 1"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="valid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Address Line 2"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {
                "color": "#282828",
              },
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Country"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Zip Code"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="City"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "height": 60,
            "marginBottom": 0,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="State"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {
                "height": 50,
              },
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 200,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="E-mail Address"
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": "100%",
          }
        }
        testID="parent"
      >
        <View
          testID="invalid"
        />
        <View
          testID="unfocused"
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          numberOfLines={1}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "borderColor": "rgba(0,0,0,0)",
                "borderRadius": 8,
                "borderWidth": 2,
                "color": "#9A9A9A",
                "flex": 1,
                "fontSize": 16,
                "padding": 10,
                "textAlignVertical": "top",
                "width": "100%",
              },
              Object {},
              Object {},
              Object {},
              Object {},
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
        <View />
        <View
          style={
            Object {
              "height": 0,
              "width": "100%",
              "zIndex": 5,
            }
          }
        >
          <View />
        </View>
      </View>
      <TouchableOpacity
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
          }
        }
        activeOpacity={0.2}
        checkedIcon={
          <Text>
            X
          </Text>
        }
        size={24}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 0,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 10,
            "width": "50%",
          }
        }
        testID="checkbox"
        uncheckedColor="#bfbfbf"
        uncheckedIcon={
          <Text>
            O
          </Text>
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text>
            O
          </Text>
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Remember Me
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={1}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3490dc",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "grey",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
            },
            Object {
              "width": "100%",
            },
            Object {},
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
            ]
          }
        >
          Register
        </Text>
      </TouchableOpacity>
    </ScrollView>
  </View>
</View>
`;
